sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala

```diff
@@ -98,7 +100,12 @@ case class ExplainCommand(
 
   // Run through the optimizer to generate the physical plan.
   override def run(sparkSession: SparkSession): Seq[Row] = try {
-    val queryExecution = sparkSession.sessionState.executePlan(logicalPlan)
+    val queryExecution =
+      if (logicalPlan.isStreaming) {
+        new IncrementalExecution(sparkSession, logicalPlan, OutputMode.Append(), "<unknown>", 0)
```

Comment:
Is this used only by the df/ds created by `spark.read.stream` (so, the mode at here does not matter?)? Maybe we should add a comment at here?